app/classes/Transvision/Search.php

```diff
@@ -0,0 +1,228 @@
+<?php
+namespace Transvision;
+
+/**
+ * Search class
+ *
+ * Allows searching for data in our repositories using a fluent interface.
+ * Currently, only the regex part (definition of the search) is implemented.
+ * ex:
```

Comment:
Trivia: I noticed it while adding the suggestion API, is it a French abbreviation? Not sure how common it is in English